### PR TITLE
Issue/199 custom 404 page

### LINKED ISSUE #199: Ability to customize error page (404 Not Found)
For now, error page just showing `Not found: /notfoundlink` instead of customizable

### PR DESCRIPTION
**Pull request by detautama**
Fixes #199 

## To use custom 404
User need to define it on `page` template like this setup
image

If 404 page didn't define, server will return 404 error text
image

## DEMO
pr-404

### CODE REVIEW

**Comment by DennisAlund** [comment]
I think that we rather can do without the creation of page. Instead if there is a template called `http404` or something. 

So it will be up to the theme developer to create the Error pages. Since those pages should most likely not be available for the content creators to access or change.

File: functions/src/index.ts [hunk 1]
```diff
@@ -37,203 +37,209 @@ export const initializeApp = configService.setConfig;
...
-    }
-
+  console.log(`GET ${request.url}`);
+  const url = request.url.replace(/^\//, ''); // Remove leading slash
+
+  const context = await getPageContextByUrl(url);
+  if (!context) {
+    // If documenpage 404  doesn't set up
+    console.log(`[HTTP 404] page not found for: ${request.url}`);
+    return response.status(404).send(`Not found: ${request.url}`);
```
Comment (DennisAlund) [inline functions/src/index.ts]: Here is where we should fetch and render the `http404` template **if** it exists in the theme templats.

**Comment by DennisAlund** [comment]
The flow should be like this

1. Request URL
2. URL was not found (`context === null` on this [line](https://github.com/oddbit/tanam/pull/213/files/1961d34d4d7f069e08b203014b5bbec89079f13f#diff-6c19df475c823c43132605decc2cf694R225))
3. Look for a theme template named `http404` (selector name).
4. If the theme had a template with that `id` then render the template and send back HTML with `404` status code
5. If there was no such template, send back `404` code with plain text message: `Not found: ${url}`

File: functions/src/index.ts (hunk 1 above)
Comment (detautama) [reply functions/src/index.ts]: Resolve on this commit https://github.com/oddbit/tanam/pull/213/commits/0160877a37bd499709a9872dd5c7ea445b482e96

**Pull request merged**